### PR TITLE
Issues #450: Report cURL status only if it’s needed and is lacking.

### LINKED ISSUE #450: Multiple "cURL" entries in the status report
The status report can list as many as 4 identical entries for "cURL", like this:

image

Three of them are from Ubercart submodules (from `uc_authorizenet_requirements()`, `uc_cybersource_requirements()`, and `uc_paypal_requirements()`; the fourth is from `simpletest_requirements()`, if `simpletest` module is installed.

It's certainly not necessary to list four identical lines, which would make the user wonder if something is amiss to cause the multiple identical rows. Also, none of them have a description field, so there's no clue in the UI where this status notice is even coming from (or that they come from four different modules).

For the Simpletest report, there should probably be a `description` entry that notes that this requirement is coming from Simpletest (and so that would be a separate core issue). For Ubercart, though, there should be a solution that is more informative and that makes sense when more than one payment module is installed (for example, `uc_authorizenet` for credit cards and `uc_paypal` for non-credit-card PayPal).

In fact, I would suggest that if cURL is installed, there's no need for a status notice at all; there's only a need if cURL is missing. So I would suggest that we modify Ubercart's `hook_requirements()` implementations to only post a `REQUIREMENT_ERROR` notice if cURL is missing, and thereby help declutter the status report page.

### PR DESCRIPTION
**Pull request by bugfolder**
Fixes https://github.com/backdrop-contrib/ubercart/issues/450.

### CODE REVIEW

File: payment/uc_authorizenet/uc_authorizenet.install [hunk 1]
```diff
@@ -12,13 +12,13 @@ function uc_authorizenet_requirements($phase) {
...
-  $requirements['uc_authorizenet_curl'] = array(
-    'title' => $t('cURL'),
-    'value' => $has_curl ? $t('Enabled') : $t('Not found'),
-  );
   if (!$has_curl) {
-    $requirements['uc_authorizenet_curl']['severity'] = REQUIREMENT_ERROR;
-    $requirements['uc_authorizenet_curl']['description'] = $t("Authorize.net requires the PHP <a href='!curl_url'>cURL</a> library.", array('!curl_url' => 'http://php.net/manual/en/curl.setup.php'));
+    $requirements['uc_authorizenet_curl'] = array(
+      'title' => $t('cURL'),
+      'value' => $has_curl ? $t('Enabled') : $t('cURL not found'),
```
Comment (argiepiano) [inline payment/uc_authorizenet/uc_authorizenet.install]: I find this confusing. This is inside an `if` statement that runs if `!$has_curl`, so why check for this again?

File: payment/uc_cybersource/uc_cybersource.install [hunk 2]
```diff
@@ -31,13 +31,13 @@ function uc_cybersource_requirements($phase) {
...
-    $requirements['uc_cybersource_curl'] = array(
-      'title' => $t('cURL'),
-      'value' => $has_curl ? $t('Enabled') : $t('Not found'),
-    );
     if (!$has_curl) {
-      $requirements['uc_cybersource_curl']['severity'] = REQUIREMENT_ERROR;
-      $requirements['uc_cybersource_curl']['description'] = $t("CyberSource's Silent Order POST requires the PHP <a href='!curl_url'>cURL</a> library.", array('!curl_url' => 'http://php.net/manual/en/curl.setup.php'));
+      $requirements['uc_cybersource_curl'] = array(
+        'title' => $t('cURL'),
+        'value' => $has_curl ? $t('Enabled') : $t('cURL not found'),
```
Comment (argiepiano) [inline payment/uc_cybersource/uc_cybersource.install]: Same comment as above.

File: payment/uc_paypal/uc_paypal.install [hunk 3]
```diff
@@ -15,13 +15,13 @@ function uc_paypal_requirements($phase) {
...
-    $requirements['uc_paypal_curl'] = array(
-      'title' => $t('cURL'),
-      'value' => $has_curl ? $t('Enabled') : $t('Not found'),
-    );
     if (!$has_curl) {
-      $requirements['uc_paypal_curl']['severity'] = REQUIREMENT_ERROR;
-      $requirements['uc_paypal_curl']['description'] = $t("PayPal WPP requires the PHP <a href='!curl_url'>cURL</a> library.", array('!curl_url' => 'http://php.net/manual/en/curl.setup.php'));
+      $requirements['uc_paypal_curl'] = array(
+        'title' => $t('cURL'),
+        'value' => $has_curl ? $t('Enabled') : $t('cURL not found'),
```
Comment (argiepiano) [inline payment/uc_paypal/uc_paypal.install]: Ditto.

File: payment/uc_authorizenet/uc_authorizenet.install (hunk 1 above)
Comment (bugfolder) [reply payment/uc_authorizenet/uc_authorizenet.install]: Of course. Chalk it up to blindness.

**Pull request merged**